src/+otp/Problem.m

```diff
@@ -27,47 +23,46 @@
         function obj = Problem(name, expectedNumVars, timeSpan, y0, parameters)
             % Constructs a problem
             
-            % Switch to class parameter validation when better supported
-            if ~ischar(name) || isempty(name)
-                error('The problem name must be a nonempty character array');
-            end
             obj.Name = name;
-            obj.ExpectedNumVars = expectedNumVars;
-            obj.Settings = struct('timeSpan', timeSpan(:), 'y0', y0, 'parameters', parameters);
+            obj.Settings = otp.ProblemSettings(expectedNumVars, timeSpan(:), y0, parameters);
         end
         
         function set.TimeSpan(obj, value)
-            obj.Settings.timeSpan = value(:);
+            obj.Settings.TimeSpan = value;
         end
         
         function timeSpan = get.TimeSpan(obj)
-            timeSpan = obj.Settings.timeSpan;
+            timeSpan = obj.Settings.TimeSpan;
         end
         
         function set.Y0(obj, value)
-            obj.Settings.y0 = value;
+            obj.Settings.Y0 = value;
         end
         
         function y0 = get.Y0(obj)
-            y0 = obj.Settings.y0;
+            y0 = obj.Settings.Y0;
         end
         
         function set.Parameters(obj, value)
-            obj.Settings.parameters = value;
+            obj.Settings.Parameters = value;
         end
         
         function parameters = get.Parameters(obj)
-            parameters = obj.Settings.parameters;
+            parameters = obj.Settings.Parameters;
         end
         
         function set.Settings(obj, value)
-            obj.validateNewState(value.timeSpan, value.y0, value.parameters);
+            expNumVars = value.ExpectedNumVars;
+            if ~(isempty(expNumVars) || length(value.Y0) == expNumVars)
```

Comment:
If I change it here, there are dozens of other places where it would need to change.  `Y0` is guaranteed to be a vector, so they are equivalent.  Most people are more familiar with `length`.